improver/utilities/temporal.py

```diff
@@ -254,3 +265,221 @@ def extract_nearest_time_point(cube, dt, time_name="time", allowed_dt_difference
+                UTC time that matches the local_time at each point.
+
+        """
+        template_cube = cube.slices_over("time").next().copy()
+        template_cube.remove_coord("time")
+        template_cube.remove_coord("forecast_period")
+        output_cube = create_new_diagnostic_cube(
+            template_cube.name(),
+            template_cube.units,
+            template_cube,
+            generate_mandatory_attributes([template_cube]),
+            optional_attributes=template_cube.attributes,
+            data=self.output_data,
+        )
+
+        # Create a local time coordinate to help with plotting data.
+        local_time_coord_standards = TIME_COORDS["time_in_local_timezone"]
+        local_time_units = cf_units.Unit(
+            local_time_coord_standards.units,
+            calendar=local_time_coord_standards.calendar,
+        )
+        timezone_points = np.array(
+            np.round(local_time_units.date2num(local_time)),
+            dtype=local_time_coord_standards.dtype,
+        )
+        output_cube.add_aux_coord(
+            AuxCoord(
+                timezone_points,
+                long_name="time_in_local_timezone",
+                units=local_time_units,
+            )
+        )
+        output_cube.add_aux_coord(
+            AuxCoord(
+                self.time_points,
+                bounds=self.time_bounds,
+                standard_name="time",
+                units=self.time_units,
+            ),
+            (0, 1),
+        )
+        return output_cube
+
+    def fill_timezones(self, input_cube, timezone_cube):
+        """
+        Populates the output cube data with data from input_cube. This is done by
+        multiplying the inverse of the timezone_cube.data with the input_cube.data and
+        summing along the time axis. Because timezone_cube.data is a mask of 1 and 0,
+        inverting it gives 1 where we WANT data and 0 where we don't. Summing these up
+        produces the result. The same logic can be used for times.
+        Modifies self.output_cube and self.time_points.
+
+        Args:
+            input_cube (iris.cube.Cube):
+                Cube of data to extract timezone-offsets from. Must contain a time
+                coord spanning all the timezones.
+            timezone_cube (iris.cube.Cube):
+                Cube describing the UTC offset for the local time at each grid location
+                Must have the same spatial coords as input_cube.
+                Cube will have a UTC_offset coord. Data will be 0 (included) or
+                1 (excluded) indicating which points are in each time zone.
+
+        Raises:
+            TypeError:
+                If combining the timezone_cube and input_cube results in float64 data.
+                (Hint: timezone_cube should be int8 and input cube should be float32)
+        """
+        bounds_offsets = self.get_time_bounds_offset(input_cube)
+        result = input_cube.data * (1 - timezone_cube.data)
```

Comment:
D'oh! I'm so stupid!! Ok. At least it was easy to fix. The unit-tests were fixed by switching the rows in the input timezone cube so that all the results stayed the same.